client/data/posts/use-create-new-post.js

```diff
@@ -0,0 +1,28 @@
+import { useCallback } from 'react';
+import { useMutation } from 'react-query';
+import wpcom from 'calypso/lib/wp';
+
+export default function useCreateNewPost( mutationOptions = {} ) {
+	const mutation = useMutation(
+		async ( { siteId, post } ) =>
```

Comment:
The downside with this open `post` approach is that we don't really define a schema of what post data is expected here, and that is prone to errors and missing/unexpected data. Should we try to introduce some sort of schema, or at least document the expected data format better?

Reply:
I've moved the file to .ts and imported the relevant Post type from `@wordpress/core-data` -- let me know if that seems like a good solution.